frontend/src/components/AddDebuggingSessionModal.vue

```diff
@@ -42,38 +38,28 @@ const missingValues = computed(() => {
 const emit = defineEmits(['createDebuggingSession'])
 
 async function createNewDebuggingSession() {
-  const mlWorkers = await api.getMLWorkerSettings();
-  const isWorkerAvailable = mlWorkers.find(
-    value => {
-      if (project.value?.mlWorkerType === 'EXTERNAL') {
-        return value.isRemote === true;
-      } else {
-        return value.isRemote === false;
-      }
-    }
-  )
-
-  if (!isWorkerAvailable) {
+  await mlWorkerStore.checkExternalWorkerConnection();
+
+  if (!mlWorkerStore.isExternalWorkerConnected) {
     useMainStore().addNotification({
       content: 'ML Worker is not connected. Please start the ML Worker first and try again.',
       color: TYPE.ERROR,
     });
-    return;
+    throw new Error('ML Worker is not connected. Please start the ML Worker first and try again.');
   }
 
   loading.value = true;
```

Comment:
We should put the loading at the start.

Reply:
I did that, but it was swift. So in the end there were always two notifications (one of loading and the second with the result) on top of another.